### PR TITLE
(gh-35) Add a stop_service parameter to install task

### PR DESCRIPTION
**Pull request by jpartlow**
This is for integration with openbolt for openvoxproject/openbolt@44.

Bolt was providing a stop_service parameter to the puppetlabs-puppet_agent module here:
https://github.com/OpenVoxProject/openbolt/blob/b9bff5a8dfe2f41218281bf8c5e6ad901bcae460/lib/bolt/plugin.rb#L133

This was for puppetlabs/bolt#1204, and while the current linux packages, at least, install with service stopped, it's probably still good practice to ensure it for the openbolt case. (This may have been more of an issue when some packages were still SysV?)

Since our install task is more general, it could be used for openvox-server/openvoxdb, but there's no particular use case for this, and I haven't bothered with acceptance tests for them.

#### This Pull Request (PR) fixes the following issues

* #35

### CODE REVIEW

File: tasks/install_linux.sh [hunk 1]
```diff
@@ -9,6 +9,7 @@ version=${PT_version:-latest}
 collection=${PT_collection:-openvox8}
 yum_source=${PT_yum_source:-https://yum.voxpupuli.org}
 apt_source=${PT_apt_source:-https://apt.voxpupuli.org}
+stop_service=${PT_stop_service:-'false'}
```
Comment (bastelfreak) [inline tasks/install_linux.sh:12]: if you default to `false` here, doesn't it make more sense to switch from `Optional[Boolean]` to `Boolean`?

Comment (jpartlow) [reply tasks/install_linux.sh:12]: The false in linux.sh is just a sane backstop for the script itself.

I think the question is more given that install.json has a default for stop_service, can the type be just Boolean. But I've had trouble with this specific corner case (puppet_library integration) in the past:

622423a525821eb48894c29481af1ce42667d632

possibly because of how the task run is constructed internally in Bolt?

And this is why apply_prep currently fails using openvox_bootstrap::install, because package is required in that corner case, despite the default:

```
jpartlow@archimedes:~/work/src/kvm_automation_tooling$ cat plans/test.pp 
plan kvm_automation_tooling::test() {
  apply_prep('b-alma9-ov8-agent-1')
# run_task('openvox_bootstrap::install', ['b-alma9-ov8-agent-1'],
#   'package' => 'openvox-agent'
# )
}
jpartlow@archimedes:~/work/src/kvm_automation_tooling$ be bolt plan run kvm_automation_tooling::test --inventory terraform/instances/inventory.b-alma9-ov8.yaml 
Bundler is using a binstub that was created for a different gem (bolt).
You should run `bundle binstub openbolt` to work around a system/bundle conflict.
Starting: plan kvm_automation_tooling::test
Starting: install puppet and gather facts on b-alma9-ov8-agent-1
Finished: plan kvm_automation_tooling::test in 0.27 sec
Failed on b-alma9-ov8-agent-1:
  Error executing plugin task from puppet_library: Task openvox_bootstrap::install:
   expects a value for parameter 'package'
Failed on 1 target: b-alma9-ov8-agent-1
Ran on 1 target
```

Which is why https://github.com/OpenVoxProject/openbolt/pull/49#issuecomment-3165590452 would be required.

Although in this case, stop_service is explicitly in the parameter list defaults Bolt is already constructing, so that shouldn't be an issue.

 But I'd rather keep the parameter list consistent, and in general, the install task, for it's base case of 'get openvox on this node now', shouldn't require anything.

Consequently I'm more inclined to actually change package type to Optional[String], even with the above pr 49 change.

But I don't think there's a perfect answer here :shrug:. Maybe the best case if for the install.json to provide the best information to the user, which I think is that all the parameters are optional and what their defaults are?

Added a commit for that.

Comment (bastelfreak) [reply tasks/install_linux.sh:12]: Ahh okay. Okay let's stick with the current implementation.

**Review by bastelfreak — APPROVED** [review]
